sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java

```diff
@@ -390,4 +391,15 @@ private static HttpRequestInitializer chainHttpRequestInitializer(
       }
     }
   }
+
+  /** GCP <a href="https://cloud.google.com/kms/">Cloud KMS</a> key for operations. */
+  @Description(
+      "GCP Cloud KMS key for creation of new objects, such as GCS objects and BigQuery tables. "
+          + "Semantics vary per service. Key format is: "
+          + "projects/<project>/locations/<location>/keyRings/<keyring>/cryptoKeys/<key>")
+  @Experimental
+  @Nullable
+  String getGcpKmsKey();
```

Comment:
I'm changing the semantics and renaming the command line flag (--gcpKmsKey). It will no longer affect IO sources and sinks, only Dataflow and bucket creation in gcpTempLocation (used to stage files for Dataflow pipelines).

For GCS, KMS can still be used using bucket default keys.